### PR TITLE
Connection to SQL DB

### PR DESCRIPTION
**Pull request by nvkaip**

### CODE REVIEW

File: src/main/java/hw18/dbService/DBService.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    public void printConnectInfo() {
+        try {
+            System.out.println("DB name: " + connection.getMetaData().getDatabaseProductName());
+            System.out.println("DB version: " + connection.getMetaData().getDatabaseProductVersion());
+            System.out.println("Driver: " + connection.getMetaData().getDriverName());
+            System.out.println("Autocommit: " + connection.getAutoCommit());
+        } catch (SQLException e) {
+            e.printStackTrace();
```
Comment (SevostyanovGerman) [inline src/main/java/hw18/dbService/DBService.java]: check last work in your group, u have the same problems

File: .idea/libraries/Maven__javax_persistence_javax_persistence_api_2_2.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<component name="libraryTable">
```
Comment (Sofasmile) [inline .idea/libraries/Maven__javax_persistence_javax_persistence_api_2_2.xml]: add this file to .gitignore

File: src/main/java/hw18/dbService/DBService.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            UsersDAO dao = new UsersDAO(connection);
+            dao.createTable();
+            dao.insertUser(usersDataSet);
+            connection.commit();
+            return dao.getUser(usersDataSet.getLogin()).getLogin();
+        } catch (SQLException e) {
+            try {
+                connection.rollback();
+            } catch (SQLException ignore) {
+            }
```
Comment (Sofasmile) [inline src/main/java/hw18/dbService/DBService.java]: add some logic here

File: src/main/java/hw18/dbService/DBService.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import hw18.dbService.dataSets.UsersDataSet;
+
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class DBService {
+
+    private static final Logger logger = LoggerFactory.getLogger(DBService.class);
```
Comment (Sofasmile) [inline src/main/java/hw18/dbService/DBService.java]:
```suggestion
    private static final Logger LOGGER = LoggerFactory.getLogger(DBService.class);
```

File: src/main/java/hw18/dbService/DBService.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                connection.setAutoCommit(true);
+            } catch (SQLException ignore) {
+                logger.error("addUser error in DBService", ignore);
+            }
+        }
+    }
+
+    public static Connection getMysqlConnection() throws SQLException {
+        try {
+            DriverManager.registerDriver((Driver) Class.forName("com.mysql.jdbc.Driver").newInstance());
```
Comment (Sofasmile) [inline src/main/java/hw18/dbService/DBService.java]: better make this string as a constant

File: src/main/java/hw18/dbService/DBService.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+            url.
+                    append("jdbc:mysql://").        //db type
+                    append("localhost:").           //host name
+                    append("3306/").                //port
+                    append("db_example?").          //db name
+                    append("user=tully&").          //login
+                    append("password=tully");       //password
+
+            System.out.println("URL: " + url + "\n");
```
Comment (Sofasmile) [inline src/main/java/hw18/dbService/DBService.java]: do we need sout here?

File: src/main/java/hw18/dbService/dao/UsersDAO.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+public class UsersDAO {
+
+    private Executor executor;
+
+    public UsersDAO(Connection connection) {
+        this.executor = new Executor(connection);
+    }
+
+    public UsersDataSet getUser(String login) throws SQLException {
+        return executor.execQuery("select * from users where login='" + login + "'", result -> {
```
Comment (Sofasmile) [inline src/main/java/hw18/dbService/dao/UsersDAO.java]:
```suggestion
        return executor.execQuery("SELECT * FROM users WHERE login='" + login + "'", result -> {
```
do it at all query

File: src/main/java/hw18/servlets/SignInServlet.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SignInServlet extends HttpServlet {
+
+    private static final Logger logger = LoggerFactory.getLogger(SignInServlet.class);
```
Comment (Sofasmile) [inline src/main/java/hw18/servlets/SignInServlet.java]:
```suggestion
    private static final Logger LOGGER = LoggerFactory.getLogger(SignInServlet.class);
```

File: src/main/java/hw18/templater/PageGenerator.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.Map;
+
+public class PageGenerator {
+
+    private static final Logger logger = LoggerFactory.getLogger(PageGenerator.class);
+    private static final String HTML_DIR = "templates";
+    private static PageGenerator pageGenerator;
+    private final Configuration cfg;
```
Comment (Sofasmile) [inline src/main/java/hw18/templater/PageGenerator.java]: rename fieds

File: .idea/libraries/Maven__javax_persistence_javax_persistence_api_2_2.xml (hunk 2 above)
Comment (nvkaip) [reply .idea/libraries/Maven__javax_persistence_javax_persistence_api_2_2.xml]: all ready did, but for some reasons it's still traced ((

File: src/main/java/hw18/dbService/DBService.java [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+            connection.setAutoCommit(false);
+            UsersDAO dao = new UsersDAO(connection);
+            dao.createTable();
+            dao.insertUser(usersDataSet);
+            connection.commit();
+            return dao.getUser(usersDataSet.getLogin()).getLogin();
+        } catch (SQLException e) {
+            try {
+                connection.rollback();
+            } catch (SQLException ignore) {
```
Comment (Sofasmile) [inline src/main/java/hw18/dbService/DBService.java]: better rename to:
```suggestion
            } catch (SQLException e) {
```

File: src/main/java/hw18/dbService/dao/UsersDAO.java [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public void insertUser(UsersDataSet usersDataSet) throws SQLException {
+        executor.execUpdate("INSERT INTO users (login, password)" +
+                " VALUES ('" + usersDataSet.getLogin() + "', '" + usersDataSet.getPassword() + "')");
+    }
+
+    public void createTable() throws SQLException {
+        executor.execUpdate("CREATE TABLE IF NOT EXISTS users " +
+                "(login varchar(256) not null, password varchar(256), primary key (login))");
```
Comment (Sofasmile) [inline src/main/java/hw18/dbService/dao/UsersDAO.java]:
```suggestion
                "(login varchar(256) not null, password varchar(256), PRIMARY KEY (login))");
```

**Pull request merged**